### ISSUE TITLE
mcfly software paper

### ISSUE THREAD

**Issue by vincentvanhees** [NLeSC/TEAM2018#12]
Title: Deep learning for time series made easy

Abstract: https://blog.esciencecenter.nl/mcfly-time-series-classification-made-easy-e47de8d29838

Editor: ? (Atze maybe? I also submitted an idea for another paper, if needed I can also be involved a bit in this one)

Note: Mcfly does not have shown impact, which may make it make challenging to demonstrate (potential) value.

**Comment by romulogoncalves** [comment]
Project moved to November Sprint.

**Comment by romulogoncalves** [comment]
@vincentvanhees @cwmeijer is this proposal still up for Team2018 November Sprint?

**Comment by cwmeijer** [comment]
I won't be participating in next sprint as I'm moving in that period.
So, yes, I think the proposal is still a good idea, but unfortunately I can't contribute.

**Comment by vincentvanhees** [comment]
I asked Sonja to investigate this while I was a away, but as far as I know that did not result in a concrete plan of action. So, I am worried now that it is too late to organize something. On top of that I have a lot of other work on my plate.

Moving forward: It may help if MT/TL could make an engineer responsible for mcfly and give him/her hours and T&D targets. It may be even better to make this person responsible for coordinating the maintenance, development, and dissemination of all machine learning tools developed at NLeSC, rather than only mcfly.

**Comment by vincentvanhees** [comment]
We probably need to update the Wageningen group about what is happening. If nobody is interested and able to continue these efforts then I think Wouter or I need to e-mail the Wageningen group and thank them for their interest but that we are not able to organise this collaboration for the foreseeable future.

**Comment by romulogoncalves** [comment]
For this Sprint @vincentvanhees is available 4 days, @florian-huber is availabel 4 days and @dafnevk will be available 2 days.

What could be done during the November Sprint with what you have right now and the above team?

**Comment by vincentvanhees** [comment]
Possibly:
- Draft paper and see how far we get.
- Turn [this](https://zenodo.org/record/1160410#.W-rWd5NKguU) public dataset into a first benchmark, because it is a simple time window classification task in line with what mcfly expects. My idea was to let the post-doc on the Lilly project explore this, but we could make a start during the sprint and when the post-doc start she can continue these efforts. I think this will simplify the organisation and provide more certainty about ongoing efforts to turn this into a paper after the sprint because we have a funded project (part of Lilly grant) to work on it. For the Wageningen collaboration this would be more uncertain.
- If we have a larger team then maybe also work on open mcfly issues.

**Comment by romulogoncalves** [comment]
@vincentvanhees I think we really need to get it out of the ground.
@dafnevk will be available 3 days and not only 2 days.

@florian-huber would it be ok for you to work on this paper?

**Comment by vincentvanhees** [comment]
Just discussed it with Dafne and it looks like we have a plan.
@romulogoncalves we may like to have one virtual machine to run the evaluation in the second half of the week, can you organise this for all the sprints collectively or should organise this ourselves?

**Comment by sonjageorgievska** [comment]
@romulogoncalves  I just read the discussion. Looks like I am not really needed here, so I would rather work on the Arena paper if that is OK.

**Comment by vincentvanhees** [comment]
On our to do list for upcoming weeks:
Talk to @cwmeijer when he is back from annual leave about doing a survey among the October2018 mcfly workshop participants, similar to the survey we did after the workshop at the VU in December2017. This could be valuable input for the paper.

**Comment by romulogoncalves** [comment]
@vincentvanhees were all the goals achieved? Could you summarize in 3 lines what was achieved? Can we consider it done and close this issue?

**Comment by vincentvanhees** [comment]
On Monday (17 December) the sprint team will meet at 10 am to discuss status and next steps. Current status:
- We drafted the paper in overleaf for SoftwareX, and think it is nearly ready for submission to the journal. Sonja, Atze, Christiaan, Dafne, Florian and I provided input. Possibly we also want to discuss with Frank and MLCV sig.
- We went through and resolved a number of (small) mcfly issues and prepared a new release, but did not do the release yet.
- @florian-huber developed an additional visualisation for mcfly, which we would like to add to both the paper and as a new feature release for mclfy.
- @cwmeijer sent out a survey to the participants of the internal mcfly workshop (Oct 2018). The result of which will be included in the impact section of the paper. 4 responses so far. @cwmeijer maybe you could send one reminder?

Conclusion: This issue cannot be closed yet, but we should be able to give you a clearer update on Monday.

**Comment by vincentvanhees** [comment]
We discussed that for now we will focus on:
- Florian + Christiaan: add visualization to mcfly and create new mcfly release.
- Dafne: to make software description section of the paper read more like a story.
- Sonja: Check that we have a brief comparison with PAMAP2 literature in the paper this to show expected performance, but also acknowledge that this is not a standardised comparison.
- Vincent + Sonja: work on impact section (include poster + elaborate on paper reference + send reminder for 2018 workshop survey invite + add new workshop survey results)
- Sonja: request copy of paper that uses mcfly + improve introduction section: More focus on mcfly as rapid prototyping tool + mcfly as a means to lower the threshold for exploring deep learning.

Everyone to embed Sonja's feedback.

Time plan:
- We all try to make progress on the above before the Christmas break.
- Catch-up on January 10th at 14:00 to assess whether the paper is ready for submission.

**Comment by vincentvanhees** [comment]
Update: I have now sent a reminder invitation for the workshop survey. Hopefully this will help to boost the response rate.

**Comment by cwmeijer** [comment]
Chris: Fix release + doi and update paper with correct release number
Florian: Create figure of architecture
Vincent: Update survey text in paper and contact Frank
Sonja: Choice of parameters. Give rationale.
All: Go through paper one more time
Dafne: Draft conclusion

**Comment by vincentvanhees** [comment]
Can we delete the acknowledgement section?
I now have written "We would like to thank Attila Reiss for depositing the PAMAP2 dataset in the UCI Machine Learning Repository." but I am not sure this is necessary.

**Comment by vincentvanhees** [comment]
Latex complains about reference:
@inproceedings{feurer_towards_2018,
	title = {Towards {Further} {Automation} in {AutoML}},
	booktitle = {{ICML} 2018 {AutoML} {Workshop}},
	author = {Feurer, M. and Hutter, F.},
	month = jul,
	year = {2018}
}
It has no page number, no doi and no url.
Can we replace this by a direct [URL](https://ml.informatik.uni-freiburg.de/papers/18-AUTOML-AutoAutoML.pdf) to the online document and call it a webpage?

**Comment by florian-huber** [comment]
I added a model architecture figure and read the entire paper again.
I have mostly made a few very tiny corrections and found that it reads well. Together with Frank's first reaction it looks to me like the manuscript is nearly ready to go!

However, I found that there is one remaining problem (which I just realized now when I was reading everything again) which is the **missing axis labels in the model visualization plot**.
For the 3 plots in the middle row one can at least guess that the x-axis is probably "accuracy". But the plot on "Learning rate and regularization rate" could be read either way. 
--> Any chance we can add those changes to mcfly before submitting? (and maybe include them in a new release 1.0.3 @cwmeijer ?)

**Comment by dafnevk** [comment]
Unfortuntaly this is (or at least was, at the time of programming those plots) not an easy problem to fix.
The 'solution' we chose is to have the values shown when you hover over the plot.
There is an issue in the mcfly repository for this: https://github.com/NLeSC/mcfly/issues/148

**Comment by florian-huber** [comment]
Maybe we could then add axes to the printed version of the figure (for the paper) ?

**Comment by vincentvanhees** [comment]
Possibly good to mention that [this (pdf)](https://www.elsevier.com/journals/softwarex/2352-7110?generatepdf=true) document on the softwarex website states that the maximum document length is 6 pages. We currently have 8, excluding references.

**Comment by florian-huber** [comment]
@romulogoncalves The manuscript was submitted today to SoftwareX.
